src/meshTools/AMIInterpolation/AMIInterpolation/AMIMethod/directAMI/directAMI.C

```diff
@@ -24,6 +24,45 @@ License
 \*---------------------------------------------------------------------------*/
 
 #include "directAMI.H"
+#include "primitiveFieldsFwd.H"
+#include "pointFieldFwd.H"
+#include "faceData.H"
+
+namespace Foam {
+    struct ShootRayFunctor : public std::binary_function<label, label, void> {
+        const pointField *tgtPoints;
```

Comment:
This should all be pointers to point and vector not pointField and vectorField. Fields are objects in CPU memory and cannot be accessed in kernels.